ptf/tests/srv6.py

```diff
@@ -241,67 +249,43 @@ def testPacket(self, pkt, sid_list):
 
         # Build expected packet from the given one...
         exp_pkt = pkt.copy()
-        # Route MAC addresses
+
+        # Route and decrement TTL
         pkt_route(exp_pkt, next_hop_mac)
-        # Decrement TTL...
         pkt_decrement_ttl(exp_pkt)
 
-        # FIXME: the P4 pipeline should calculate correct checksum
-        if TCP in pkt:
-            pkt[TCP].chksum = 1
-            exp_pkt[TCP].chksum = 1
-        if UDP in pkt:
-            pkt[UDP].chksum = 1
-            exp_pkt[UDP].chksum = 1
-        if ICMPv6Unknown in exp_pkt:
-            pkt[ICMPv6Unknown].cksum = 1
-            exp_pkt[ICMPv6Unknown].cksum = 1
+        # FIXME: update P4 program to calculate correct checksum
+        set_cksum(pkt, 1)
+        set_cksum(exp_pkt, 1)
 
         testutils.send_packet(self, self.port1, str(pkt))
         testutils.verify_packet(self, exp_pkt, self.port2)
 
 
 @group("srv6")
-class FabricSrv6EndTest(P4RuntimeTest):
-    """Tests SRv6 end behavior"""
-
-    def buildSrv6Pkt(self, pkt, sid_list):
-        sid_len = len(sid_list)
-        # Build srv6 packet from the given one...
-        srv6_pkt = pkt.copy()
-        # Route MAC addresses
-        pkt_route(srv6_pkt, SWITCH1_MAC)
-        # Set IPv6 dst to first SID...
-        srv6_pkt[IPv6].dst = SWITCH1_IPV6
-        # Insert SRv6 header between IPv6 header and payload
-        srv6_hdr = IPv6ExtHdrSegmentRouting(
-            nh=pkt[IPv6].nh,
-            addresses=sid_list[::-1],
-            len=sid_len * 2,
-            segleft=sid_len,
-            lastentry=sid_len - 1)
-        srv6_pkt[IPv6].nh = 43  # next header is SR header
-        srv6_pkt[IPv6].payload = srv6_hdr / pkt[IPv6].payload
-        return srv6_pkt
+class Srv6EndTest(P4RuntimeTest):
+    """Tests SRv6 end behavior (without pop), where the switch forwards the
+    packet to the next SID found in the SRv6 header.
+    """
 
     def runTest(self):
+        my_sid = SWITCH2_IPV6
         sid_lists = (
             [SWITCH2_IPV6, SWITCH3_IPV6, HOST2_IPV6],
-            [SWITCH2_IPV6, HOST2_IPV6],
+            [SWITCH2_IPV6, SWITCH3_IPV6, SWITCH4_IPV6, HOST2_IPV6],
```

Comment:
The `MAX_HOPS` in headers.p4 is 3, not sure why this is working

Reply:
You can find error from bmv2 log
[13:23:55.679] [bmv2] [E] [thread 4367] [3.0] [cxt 0] Exception while parsing: StackOutOfBounds